data_steward/common.py

```diff
@@ -42,11 +42,32 @@
 # Values for ID fields for each HPO are summed with a factor of ID_CONSTANT_FACTOR
 ID_CONSTANT_FACTOR = 1000000000000000
 
+KEY_CONSTANT_FACTOR = 1000000000000
+
 # Added to value in all ID fields in records coming from the RDR
 RDR_ID_CONSTANT = ID_CONSTANT_FACTOR
 
+# Starting factor to create ID space for person to observation mapped records
+EHR_PERSON_TO_OBS_CONSTANT = 2 * ID_CONSTANT_FACTOR
+
 # Starting factor to create ID spaces for each HPO (without RDR collisions)
-EHR_ID_MULTIPLIER_START = 2
+EHR_ID_MULTIPLIER_START = 3
+
+# person to observation (pto) constants
+
+pto_concept_id = dict()
+# concept ids
+pto_concept_id['gender'] = 4135376
```

Comment:
I think I've come to regret every instance where I've used logic to assign constants (e.g. the [very first constant in this file](https://github.com/all-of-us/curation/blob/develop/data_steward/common.py#L4)). Please do not repeat my mistakes. :) There are several dogmatic arguments (e.g. not test-friendly), but I facepalm hardest when the logic which surrounds my constants finds its way back into the code which is _using_ the constants. :man_facepalming: I now aspire for my constants to be little more than substitutes for literal values. Please 

It's a good idea to group the constants needed for the person-to-observation step, but as it is the dict `pto_concept_id` requires extra care to use. The keys for the dict are themselves literals which we may find ourselves [mis]typing in the future. It may also have limited reusability. For example, should we need the concept id for gender elsewhere, it doesn't provide an easy way for us to get that as compared to multiple constant variables (e.g. `GENDER_CONCEPT_ID=4135376`, `RACE_CONCEPT_ID=4013886`).

The enumeration below [smells](https://martinfowler.com/bliki/CodeSmell.html) a little like my facepalm moment- I'm having difficulty following how the [client code in ehr_union_test](https://github.com/all-of-us/curation/pull/136/files#diff-5e001260be368659bdb3a2b698df77d0R296) works. Could this be replaced with multiple assignments? I may also be missing something here- please let me know if we should discuss this further in person.

Reply:
Thanks @mark-velez. I was looping through them because I was reusing code but it's best to keep them explicit for several reasons like you mentioned.